### PR TITLE
[BTS-1402] Don't apply parallelize gather rule for one-shard DBs

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

Enterprise companion: https://github.com/arangodb/enterprise/pull/1270

- [x] :hankey: Bugfix

### Checklist

- [x] Tests
  - [x] **Regression tests** - see enterprise companion
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*

#### Related Information

- Jira ticket: https://arangodb.atlassian.net/browse/BTS-1402

### CODE REVIEW

**Review by jsteemann — APPROVED** [review]
LGTM

File: arangod/Aql/OptimizerRules.cpp [hunk 1]
```diff
@@ -8348,6 +8348,11 @@ void arangodb::aql::parallelizeGatherRule(Optimizer* opt,
                                           OptimizerRule const& rule) {
   TRI_ASSERT(ServerState::instance()->isCoordinator());
 
+  if (plan->getAst()->query().vocbase().isOneShard()) {
```
Comment (mchacki) [inline arangod/Aql/OptimizerRules.cpp:8351]: Does it hurt if it is applied? Or is this "just" unnecessary work?

If yes:
If we  are in a sharded database "by accident" have collections with only 1 shard, that are all on the same server we may apply a plan that looks similar to a oneShardDB plan.

**Review by mchacki — APPROVED** [review]
In general LGTM.
Im unsure if this is enough or if we need a protection for "accidental" one shards.

**Review by mchacki — CHANGES_REQUESTED** [review]
Wait a second.
Can we have a test similar to the example in the BTS ticket?
Otherwise I cannot be sure that we actually fix something, I am not fully convinced that parallel gather causes the BTS query to "fail"

**Comment by mpoeter** [comment]
As it turned out the referenced BTS issue has a different root cause - thus closing this.

**Pull request closed without merging**